plugins/module_utils/ec2.py

```diff
@@ -1507,3 +1610,21 @@ def normalize_ec2_vpc_dhcp_config(option_config: List[Dict[str, Any]]) -> Dict[s
                 config_data[option] = [val["Value"] for val in config_item["Values"]]
 
     return config_data
+
+
+def determine_iam_arn_from_name(module, name_or_arn: str) -> str:
+    if validate_aws_arn(name_or_arn, service="iam", resource_type="instance-profile"):
+        return name_or_arn
+    iam = module.client("iam", retry_decorator=AWSRetry.jittered_backoff())
+    try:
+        return iam.get_instance_profile(InstanceProfileName=name_or_arn, aws_retry=True)["InstanceProfile"]["Arn"]
+    except is_boto3_error_code("NoSuchEntity") as e:
+        module.fail_json_aws(e, msg=f"Could not find IAM instance profile {name_or_arn}")
+    except (
+        botocore.exceptions.ClientError,
+        botocore.exceptions.BotoCoreError,
+    ) as e:  # pylint: disable=duplicate-except
+        module.fail_json_aws(
+            e,
+            msg=f"An error occurred while searching for IAM instance profile {name_or_arn}. Please try supplying the full ARN.",
+        )
```

Comment:
1) Why is this in ec2.py?  It's a pure IAM thing
1.b) it's basically a duplication of `list_iam_instance_profiles(client, name=XXX)`

2) I really dislike passing module about